### PR TITLE
x-pack/filebeat/input/entityanalytics/provider/activedirectory: new package

### PR DESCRIPTION
**Pull request by efd6**
## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ] Testing requires setting up an active directory test server (or pointing the test at an existing server). There are notes in the test file explaining how pointing the tests at the server is handled. I used instructions [here](https://albusbit.com/blog/how-to-set-up-active-directory-test-environment/) for setting up a local VM test server.
## Related issues

- For elastic/integrations#8559
- Depends #37854

### CODE REVIEW

File: x-pack/filebeat/docs/inputs/input-entity-analytics.asciidoc [hunk 1]
```diff
@@ -27,14 +28,186 @@ the <<{beatname_lc}-input-{type}-common-options>> described later.
...
+==== Active Directory (`activedirectory`)
+
+The `activedirectory` provider allows the input to retrieve users, with group
+memberships, from Active Directory.
+
+[float]
+==== Setup
+
+A user with appropriate permissions must be set up in the Active Directory
+Server Manages in order for the provider to function properly.
```
[the remark this answers is not in the document]
Comment (efd6) [reply x-pack/filebeat/docs/inputs/input-entity-analytics.asciidoc:49]: Thanks, you are correct.

File: x-pack/filebeat/input/entityanalytics/provider/activedirectory/statestore.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	if err != nil && !errIsItemNotFound(err) {
+		return nil, fmt.Errorf("unable to get last sync time from state: %w", err)
+	}
+	err = s.tx.Get(stateBucket, lastUpdateKey, &s.lastUpdate)
+	if err != nil && !errIsItemNotFound(err) {
+		return nil, fmt.Errorf("unable to get last update time from state: %w", err)
+	}
+	err = s.tx.Get(stateBucket, whenChangedKey, &s.whenChanged)
+	if err != nil && !errIsItemNotFound(err) {
+		return nil, fmt.Errorf("unable to get last change time from state: %w", err)
```
[the remark this answers is not in the document]
Comment (efd6) [reply x-pack/filebeat/input/entityanalytics/provider/activedirectory/statestore.go:82]: No, what is there is correct.

File: x-pack/filebeat/docs/inputs/input-entity-analytics.asciidoc [hunk 3]
```diff
@@ -27,14 +28,186 @@ the <<{beatname_lc}-input-{type}-common-options>> described later.
...
+
+[float]
+===== `ad_base_dn`
+
+The Active Directory Base Distinguished Name. Field is required.
+
+[float]
+===== `ad_user`
+
+The client user name. Used for authentication. Field is required.
```
[the remark this answers is not in the document]
Comment (efd6) [reply x-pack/filebeat/docs/inputs/input-entity-analytics.asciidoc:190]: Will add.

File: x-pack/filebeat/input/entityanalytics/provider/activedirectory/activedirectory.go [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+	if err != nil {
+		return fmt.Errorf("unable to commit state: %w", err)
+	}
+
+	return nil
+}
+
+// runIncrementalUpdate will run an incremental update. The process is similar
+// to full synchronization, except only users which have changed (newly
+// discovered, modified, or deleted) will be published.
```
[the remark this answers is not in the document]
Comment (efd6) [reply x-pack/filebeat/input/entityanalytics/provider/activedirectory/activedirectory.go:238]: AD doesn't have a notion of deleted. I was thinking about this and wasn't sure what we should do. At the very least, "deleted" should be removed from the docs, but I'll take a look at possible approaches to see if we can reconstruct this functionality.

[the remark this answers is not in the document]
Comment (efd6) [reply x-pack/filebeat/input/entityanalytics/provider/activedirectory/activedirectory.go:238]: I've added a method for handling deletion. It's not a complete approach, but I think it is the best we can get.

[the remark this answers is not in the document]
Comment (kcreddy) [reply x-pack/filebeat/input/entityanalytics/provider/activedirectory/activedirectory.go:238]: Thanks Dan. That makes sense

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**